packages/bridge-ui-v2/src/components/Bridge/Amount.svelte

```diff
@@ -53,8 +59,38 @@
       !$destNetwork ||
       !$tokenBalance ||
       $enteredAmount === BigInt(0) // no need to check if the amount is 0
-    )
+    ) {
+      $insufficientBalance = false;
+      return;
+    }
+
+    if ($enteredAmount > $tokenBalance.value) {
+      $insufficientBalance = true;
+      // no need to check any further if the amount is greater than the balance
       return;
+    }
+    $insufficientBalance = false;
+
+    const erc20Bridge = bridges.ERC20 as ERC20Bridge;
+    const spenderAddress = chainContractsMap[$network.id].tokenVaultAddress;
+
+    const tokenAddress = await getAddress({
+      token,
+      srcChainId: $network.id,
+      destChainId: $destNetwork.id,
+    });
+
+    if (!tokenAddress) return;
+
+    // Check for allowance
+    $insufficientAllowance = await erc20Bridge.requireAllowance({
```

Comment:
Why do we need to do this here?, we already get the error when running the estimation, don't we?, is this not redundant?

Reply:
As discussed in the call, the gas estimation does not throw an error, so we need to call it.